src/Event/DynamoDb/DynamoDbRecord.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Event\DynamoDb;
+
+final class DynamoDbRecord
+{
+    /** @var array */
+    private $record;
+
+    /**
+     * @param mixed $record
+     */
+    public function __construct($record)
+    {
+        if (! is_array($record) || ! isset($record['eventSource']) || $record['eventSource'] !== 'aws:dynamodb') {
+            throw new \InvalidArgumentException('Event source must come from DynamoDB');
+        }
+
+        $this->record = $record;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getKeys()
+    {
+        return $this->record['dynamodb']['Keys'];
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getNewImage()
+    {
+        return $this->record['dynamodb']['NewImage'];
+    }
+
+    /**
+     * @return mixed|null
+     */
+    public function getOldImage()
+    {
+        return $this->record['dynamodb']['OldImage'] ?? null;
+    }
+
+    public function getSequenceNumber(): string
+    {
+        return $this->record['dynamodb']['SequenceNumber'];
+    }
+
+    public function getSizeBytes(): int
+    {
+        return $this->record['dynamodb']['SizeBytes'];
+    }
+
+    public function getStreamViewType(): string
+    {
+        return $this->record['dynamodb']['StreamViewType'];
```

Comment:
I had a quick look at what this field means: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_StreamSpecification.html

It seems some DynamoDb events can only contain keys, or new images, or old images.

Does that mean that we should make all these 3 getters nullable above?

Reply:
Keys should not be nullable, dynamodb always requires/returns keys